src/base/random.test.cc

```diff
@@ -0,0 +1,161 @@
+ * All rights reserved
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met: redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer;
+ * redistributions in binary form must reproduce the above copyright
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution;
+ * neither the name of the copyright holders nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <gtest/gtest-spi.h>
+#include <gtest/gtest.h>
+
+#include <unordered_map>
+
+#include "base/gtest/logging.hh"
+#include "base/random.hh"
+
+using namespace gem5;
+
+/**
+ * Checking that default construction uses the default
+ * seed as specified by the standard
+ */
+TEST(RandomCtor, DefaultConstruct)
+{
+    // Init seed was default seed 5489
+    Random dut;
+    // First number corresponds to the default seed
+    ASSERT_EQ(dut.random<uint64_t>(), 14514284786278117030llu);
+}
+
+
+/**
+ * Checking that default construction uses the default
+ * see as specified by the standard
+ */
+TEST(RandomCtor, ConstructUserSpecifiedSeed)
+{
+    Random dut{42};
+    ASSERT_EQ(dut.random<uint64_t>(), 13930160852258120406llu);
+}
+
+/**
+ * Test that reseeding after construction with
+ * an explicit seed works
+ */
+TEST(RandomCtor, ConstructThenReseed)
+{
+    Random dut{};
+    ASSERT_EQ(dut.random<uint64_t>(), 14514284786278117030llu);
+    dut.init(42);
+    ASSERT_EQ(dut.random<uint64_t>(), 13930160852258120406llu);
+}
+
+/**
+ * This tests check that range generation
+ * works if min equals max
+ */
+TEST(RandomRange, MinEqualsMax)
+{
+    Random dut;
+
+    for (int i = 0; i < 10; i++) {
+        ASSERT_EQ(dut.random<int>(0, 0), 0);
+        ASSERT_EQ(dut.random<uint64_t>(1, 1), 1);
+        ASSERT_EQ(dut.random<int>(-1, -1), -1);
+    }
+}
+
+constexpr int loopCount = 30000;
+
+bool within_freq_range(int count)
```

Comment:
The coding style dictates that functions should be in camel case.

Also, could you add documentation to this function? It is clear to me what it does, but it may not be to a future reader